src/service/git/github/github-client.ts

```diff
@@ -37,16 +37,23 @@ export default class GitHubClient implements GitClient {
     return "noreply@github.com";
   }
 
-  async getPullRequest(owner: string, repo: string, prNumber: number, squash = true): Promise<GitPullRequest> {
+  async getPullRequest(owner: string, repo: string, prNumber: number, squash: bool | undefined): Promise<GitPullRequest> {
     this.logger.debug(`Fetching pull request ${owner}/${repo}/${prNumber}`);
     const { data } = await this.octokit.rest.pulls.get({
       owner: owner,
       repo: repo,
       pull_number: prNumber,
     });
 
+    const sha = this.getSha(pr);
+    if (squash === undefined) {
+       squash = sha is not a merge commit;
```

Comment:
How can you evaluate this `sha is not a merge commit` ? that is what I am missing right now.